packages/opentelemetry-resources/src/platform/node/service-name.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { existsSync, readFileSync } from 'fs';
+import { join } from 'path';
+import { cwd } from 'process';
+
+export function serviceName(): string {
+  return getPJsonName() ?? getExecPathName();
+}
+
+function getPJsonName(): string | undefined {
```

Comment:
can you please add here some more description about which package.json it should get, either the otel or the the package that will be using otel etc. ?

Getting this always at runtime might be tricky with all different edge cases, are we actually even able to test that (running form different package, running from npm command, from different folder with just long path using for example pm2 etc. etc.). 
I'm a bit sceptical to do it at runtime, it would be safer to generate this during build / watch and simply include it the same we do with `version.ts` so that you don't have to search this on drive - there is similar things in different PR, I'm just affraid we might be not aware of all edge cases. We could simply export 2 things from version.ts `VERSION` and `PACKAGE_NAME` for example?

```
export const PACKAGE_VERSION = '${pjson.name}';
```
It will be bundled and you don't need to make sync blocking  call, WDYT ?

Reply:
We can't include it at build time because it is the customer's package.json. That's why it uses `cwd`.

Reply:
if we don't think it would be reliable, maybe we should just use a hardcode value for the default service name and require users to set the env var.

Reply:
I don't think it cost much to use it if we have it available, that will always be better than a random one.
In the case of pm2, the argv0 is not the app but some sort of "container" that then require the actual app, so argv0 is definitly not reliable.